src/main/java/io/codelex/flightplanner/LogisticServiceFromDB.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+@Service
+@ConditionalOnProperty(prefix = "myapp", name = "appmode", havingValue = "database")
+public class LogisticServiceFromDB implements LogisticService {
+
+    private final LogisticRepositoryAirport logisticRepositoryAirport;
+    private final LogisticRepositoryFlight logisticRepositoryFlight;
+
+    public LogisticServiceFromDB(LogisticRepositoryAirport logisticRepositoryAirport, LogisticRepositoryFlight logisticRepositoryFlight) {
+        this.logisticRepositoryFlight = logisticRepositoryFlight;
+        this.logisticRepositoryAirport = logisticRepositoryAirport;
+    }
+
+    @Override
+    public void clearFlights() {
+        logisticRepositoryFlight.deleteAll();
+    }
+
+    @Override
+    public synchronized Flight addFlight(Flight flight) {
+        if (checkSameFlight(flight)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        } else {
+            if (checkSameAirport(flight)) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+            } else {
+                if (checkTime(flight)) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+                } else {
+                    logisticRepositoryAirport.save(flight.getFrom());
+                    logisticRepositoryAirport.save(flight.getTo());
+                    logisticRepositoryFlight.save(flight);
+                    return flight;
+                }
+            }
+        }
+    }
+
+    @Override
+    public List<Flight> getFlight() {
+        return logisticRepositoryFlight.findAll();
+    }
+
+    @Override
+    public Flight fetchFlight(Integer id) {
+        Flight returnFlight = logisticRepositoryFlight.findById(id).orElse(null);
+        if (returnFlight == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        return returnFlight;
+    }
+
+    @Override
+    public synchronized void deleteFlight(Integer id) {
+        if (logisticRepositoryFlight.existsById(id)) {
+            logisticRepositoryFlight.deleteById(id);
+        }
+    }
+
+    @Override
+    public SearchItems searchFlight(FlightsRequest flightsRequest) {
+        if (flightsRequest.getFrom() != null && flightsRequest.getTo() != null && flightsRequest.getDepartureDate() != null) {
+            if (flightsRequest.getFrom().equals(flightsRequest.getTo())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+            } else {
+                List<Flight> items = logisticRepositoryFlight.findAll().stream().filter(fl -> fl.getFrom().getAirport().equals(flightsRequest.getFrom()) &&
```

Comment:
Vot šitā nevajadzētu darīt:
flightRepo.findAll()
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries

Tas pats arī zemāk iekš searchAirports un iekš checkSameFlight